x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authz/RoleDescriptorTests.java

```diff
@@ -538,12 +668,40 @@ private RoleDescriptor randomRoleDescriptor() {
         }
         final ConfigurableClusterPrivilege[] configurableClusterPrivileges;
         if (randomBoolean()) {
-            configurableClusterPrivileges = new ConfigurableClusterPrivilege[] {
-                new ConfigurableClusterPrivileges.ManageApplicationPrivileges(
-                    Sets.newHashSet(generateRandomStringArray(3, randomIntBetween(4, 12), false, false))
-                ) };
+            if (randomBoolean()) {
+                configurableClusterPrivileges = new ConfigurableClusterPrivilege[0];
+            } else {
+                configurableClusterPrivileges = new ConfigurableClusterPrivilege[] {
+                    new ConfigurableClusterPrivileges.ManageApplicationPrivileges(
+                        Sets.newHashSet(generateRandomStringArray(3, randomIntBetween(4, 12), false, false))
+                    ) };
+            }
         } else {
-            configurableClusterPrivileges = new ConfigurableClusterPrivilege[0];
+            if (randomBoolean()) {
+                configurableClusterPrivileges = new ConfigurableClusterPrivilege[] {
+                    new ConfigurableClusterPrivileges.UpdateProfileDataPrivileges(
+                        Sets.newHashSet(generateRandomStringArray(3, randomIntBetween(4, 12), false, false))
+                    ) };
+            } else {
+                if (randomBoolean()) {
+                    configurableClusterPrivileges = new ConfigurableClusterPrivilege[] {
+                        new ConfigurableClusterPrivileges.UpdateProfileDataPrivileges(
+                            Sets.newHashSet(generateRandomStringArray(3, randomIntBetween(4, 12), false, false))
+                        ),
+                        new ConfigurableClusterPrivileges.ManageApplicationPrivileges(
+                            Sets.newHashSet(generateRandomStringArray(3, randomIntBetween(4, 12), false, false))
+                        ) };
+                } else {
+                    configurableClusterPrivileges = new ConfigurableClusterPrivilege[] {
+                        new ConfigurableClusterPrivileges.ManageApplicationPrivileges(
+                            Sets.newHashSet(generateRandomStringArray(3, randomIntBetween(4, 12), false, false))
+                        ),
+                        new ConfigurableClusterPrivileges.UpdateProfileDataPrivileges(
+                            Sets.newHashSet(generateRandomStringArray(3, randomIntBetween(4, 12), false, false))
+                        ) };
+
+                }
+            }
```

Comment:
Same suggestion here for switch block vs nested if/else.

Reply:
I have replaced nested ifs with a switch expression.